test/validation/api/traffic_mngr/traffic_mngr.c

```diff
@@ -1098,6 +1098,46 @@ static int make_pkts(uint32_t    num_pkts,
 	return 0;
 }
 
+static uint32_t send_pkts_multi(odp_tm_queue_t tm_queue, uint32_t num_pkts)
+{
+	xmt_pkt_desc_t *xmt_pkt_desc;
+	odp_packet_t    odp_pkt;
+	uint32_t        xmt_pkt_idx, pkts_sent;
+	int64_t         rc, i = 0;
+
+	/* Now send the pkts as fast as we can. RED drops are internally
+	 * consumed by odp_tm_enq_multi().
+	 */
+	xmt_pkt_idx = num_pkts_sent;
+	rc = odp_tm_enq_multi(tm_queue, &xmt_pkts[xmt_pkt_idx], num_pkts);
+
+	/* Record consumed packets */
+	pkts_sent = 0;
+	for (i = 0; i < rc; i++) {
+		xmt_pkt_desc = &xmt_pkt_descs[xmt_pkt_idx + i];
+		xmt_pkt_desc->xmt_idx = xmt_pkt_idx + i;
+		xmt_pkt_desc->xmt_time = odp_time_local();
+		xmt_pkt_desc->tm_queue = tm_queue;
+		pkts_sent++;
+	}
+
+	/* Free rejected pkts */
+	if (rc < num_pkts) {
+		i = rc > 0 ? rc : 0;
+		for (i = rc; i < num_pkts; i++) {
```

Comment:
Here in the for-statement i gets assigned again, potentially to a negative value. Just removing that should work, but maybe you could just rely on the fact that i has the correct value after the previous for loop and this loop is just a continuation of that. So maybe you could do this:

````

+               pkts_sent++;
+       }
+       /* Free rejected pkts */
+       for (; i < num_pkts; i++) {
+               xmt_pkt_desc = &xmt_pkt_descs[xmt_pkt_idx + i];
````

Otherwise this PR looks ok. I think you could add my review tag already in the next version.